tests/vector.rs

```diff
@@ -90,9 +89,8 @@ macro_rules! impl_test_rem {
 
 macro_rules! impl_test_iter_sum {
     ($VectorN:ident { $($field:ident),+ }, $ty:ty, $s:expr, $v:expr) => (
-        let res: $VectorN<$ty> = iter::repeat($v).take($s as usize).sum();
-        assert_eq!(res,
-                   $VectorN::new($($v.$field * $s),+));
+        assert_eq!($VectorN::new($($v.$field * $s),+),
+                   iter::repeat($v).take($s as usize).sum());
```

Comment:
Well... more that it could not deduce the type parameter of the `sum` method symmetrically. The other solution would be:

```rust
assert_eq!(iter::repeat($v).take($s as usize).sum::<$VectorN<_>>(),
                  $VectorN::new($($v.$field * $s),+));
```

But I found swapping them to look more pleasing, despite not following the pattern of putting the value being tested first...